File: String-Algorithms/String-Algorithms-master/URLify/URLify.py
class solution:
    def URLify(self,input_str,key):
        input2 = ""
        for i in range(len(input_str)):
            if(input_str[i] != ' '):
                input2+=input_str[i]
            elif((input_str[i]==' ') and (input_str[i+1] == ' ')):
                return input2
            elif((input_str[i]==' ') and (input_str[i+1] != ' ')):
                input2 += key
        return input2

input1 = str(input())
key = str(input())
sol = solution()
print(sol.URLify(input1,key))

File: String-Algorithms/String-Algorithms-master/URLify/URLify.cpp
# include<iostream>
# include<cstdlib>
# include<string>
# include<map>

using namespace std;

class solution
{
public:
    string URLify(string input,string key)
    {
        string input2;
        int size_string = input.size();
        for(int i=0;i<size_string;i++)
        {
            if(input[i]!=' ')
            {
                input2+=input[i];
            }
            else if((input[i]==' ') && (input[i+1]==' '))
            {
               return input2;
            }
            else if((input[i]==' ') && (input[i+1]!=' '))
            {
                input2 = input2 + key;
            }
        }
        return input2;
    }
};

int main()
{
    string input1,key;
    getline(cin,input1);
    cin >> key;
    solution sol;
    cout << sol.URLify(input1,key)<< endl;
    return 0;
}
